server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1303,6 +1306,36 @@ private static boolean isFiltered(Property[] properties) {
         return properties != null && Arrays.asList(properties).contains(Property.Filtered);
     }
 
+    /**
+     * A writeable validator able to check the value of string type custom setting by using regular expression
+     */
+    public static class RegexValidator implements Writeable, Validator<String> {
+        private Pattern pattern;
+
+        public RegexValidator(String regex) {
+            this.pattern = Pattern.compile(regex);
+        }
+
+        public RegexValidator(StreamInput in) throws IOException {
+            this.pattern = Pattern.compile(in.readString());
+        }
+
+        @Override
+        public void validate(String value) {
+            if (pattern.matcher(value).matches()) {
+                if (value != null) {
+                    throw new IllegalArgumentException("Setting must be valid [" + value + "]");
+                }
+                throw new IllegalArgumentException("Custom setting matches regex [" + pattern.pattern() + "]");
```

Comment:
These two `throw` statements don't seem to make sense.  If it's not null (the usual case) they just see that it's not valid, but aren't told why.  You'd only get to the second `throw` if the value was null, in which case it would not match the regex.

I think you just need a single `throw` statement (the second one).  If you want to add the value to the string you can do that to combine the two strings, but make only one throw.